### ISSUE TITLE
Tim Wilbrink appears under Login Items

### ISSUE THREAD

**Issue by kariotakisomiros** [OpenSC/OpenSC#2739]
Is it the case that since version 0.19 of opensc, in a MacBook, the following Tim Wilbrink appears under Login Items?

If so, would it be possible to ask who owns that process entry and who manages the versions of it?

Hint: its details are visible via command: `sfltool dumpbtm`

**Comment by dengert** [comment]
Google for: Tim Wilbrink
This sounds like what you see:
https://discussions.apple.com/thread/254570061
He also has a number of apps. 

Did you install OpenSC?

Why do you think this is related to OpenSC?

Maybe one of his apps uses OpenSC?

**Comment by frankmorgner** [comment]
Tim Wilbrink sponsors the macOS signing for OpenSC. His developer certificate is the signing entity in our CI process.

**Comment by frankmorgner** [comment]
Maybe there's some confusion with the file names... I'll need to check as soon as I get my hands on a macbook...

**Comment by karimbenbourenane** [comment]
Could you please change the certificate to something with a signing entity name that makes sense? Everyone that has installed OpenSC has what appears to be a seemingly random background login item in their System Preferences for "Tim Wilbrink" and that is sure to confuse a lot of users.

**Comment by fgeorgatos** [comment]
there is more to it:
* the `opensc` source code has no reference to show how and at which point that login item is defined
* the ci/cd process is presumably incomplete to show what happens if this item gets disabled (read: it seems nothing)
* the code ownership about the signing process itself is of unknown provenance, its versioning too.
* the effect isn't visible in all macosx installations of opensc, unclear why
* also, the named developer does not appear to be a project member which isn't too reassuring either.

In short, it could also be called a Bill Gates login item and noone would be wiser about where it comes from or what it does.

The original discussions.apple.com reference shows the following picture:

<img width="503" alt="315f6003-1600-4e1c-9b55-ade3faf43590" src="https://github.com/OpenSC/OpenSC/assets/1531968/db5aaee9-c03c-4b06-80e4-dc9e6d066286">

This is a surprising situation for a security component which some people would rather use to *decrease* supply chain accident probability.

**Comment by thealanberman** [comment]
Couldn't help myself. 😏 
![tim-wilbrink-opensc](https://github.com/OpenSC/OpenSC/assets/796535/7cf702ac-db8a-4e5d-8367-50b436d7d21d)

**Comment by matglas** [comment]
I want to bump this with the general question on the governance structure of OpenSC. Is there somewhere this is described?

**Comment by karimbenbourenane** [comment]
I would like to second your bump @matglas. It has been over a year and we have no response from either Tim Wilbrink (@TWilb) or one of his colleagues -- really anyone that has contributed to or maintains OpenSC -- that can answer why his name is still stuck in my macOS Login Items to enable a piece of software that supposedly increases the security of my VPN sessions and SSH logins by enabling me to use my hardware token. Yes, it's true, the software works, but it's unnerving that it has a personal signature on it. Every year or so when I do a full security audit of my development environment I have to keep a mental footnote of this oddity, and no one should be having to do that. @TWilb can you weigh in on this? Are you planning to keep your name on this certificate indefinitely, or are there plans to change it so as to not confuse novice users that are required to use OpenSC to utilize their physical security tokens?

**Comment by fgeorgatos** [comment]
Agreed. So won't the real @TWilb please stand up?

**Comment by metsma** [comment]
It is the nature of macOS signing.
Tim probably donated his personal signing certificate to OpenSC (need to verify by git history).
Apple does not give choice to select preferred signing certificate name.
As far I know OpenSC does not have any organisation body to apply macOS development account (@frankmorgner can you correct me?).

**Comment by frankmorgner** [comment]
>  As far I know OpenSC does not have any organisation body to apply macOS development account

Yes, this is correct.

As stated above, https://github.com/OpenSC/OpenSC/issues/2739#issuecomment-1501184968, the signing process is fully automated without manual interaction from Tim. Every dmg you're pulling from our reseases was generated in our CI pipeline.

Unfortunately, entering Apple's walled garden is very burdonsome. Unless there is someone willing to invest into this (time, money) or some service handling the Apple specific parts (we have Signpath.io doing the Windows stuff), I don't think the situation will change.

**Comment by matglas** [comment]
Would something like the OpenSSF or OWASP be able to be of help?

**Comment by frankmorgner** [comment]
The nearest service that OpenSSF offers, is [Cosign/Sigstore](https://docs.sigstore.dev/signing/quickstart/), which doesn't seem to offer macOS (Apple) specific code signing; OWASP doesn't seem to offer code signing at all.

**Comment by matglas** [comment]
I reached out on the OpenSSF slack to ask what they offer.

> Projects are OpenSSF Technical Initiatives that support the innovative delivery of security tooling and best practices to secure critical open source software.

There is a long list of current projects. But Michael Lieberman, members of the TAC, did respond on my inquiry with:

> And as far as I’m aware OpenSSF doesn’t provide MacOS signing certs, but it’s something that could be brought up to as a proposal to the TAC/GB.

I don't know if maintainers behind OpenSC would be up for a conversation with them. I myself am not in the capacity to drive that conversation but I can imagine it would be benificial for more OSS projects and OpenSC is good example.

**Comment by frankmorgner** [comment]
Thanks for investigating this. If there is some ready-to-use solution, we would be grateful to participate on this. However, we are well aware that setting up such a process is not easy, in particular given the proprietary tool chain on macOS. You can contact me for information about the typical code signing process that I am using on my machine (and which I partly integrated into CI), but I don't have much time setting up a bigger background infrastructure for this.

**Comment by matglas** [comment]
Yes that is a hard one. I have experience too with Mac signing and creating an infrastructure that can be used for these types of things is not the most straight forward. I'm not sure how easy it is to move something like this forward. I do know that the OSSF is open to the idea and discussing it if its brought up in the TAC. But I personally do not have enough room to really address that in a structural way. "time constraints"

**Comment by frankmorgner** [comment]
Awareness has been raised with the last two releases. Currently, there is no plan to change the macOS signing process.

**Comment by fgeorgatos** [comment]
Kindly leave this open until the original request gets treated. Otherwise,
people will consider this as "normal".

**Comment by frankmorgner** [comment]
Sorry, but this is normal... as described above: https://github.com/OpenSC/OpenSC/issues/2739?notification_referrer_id=NT_kwDOABNTkbMxMzU1NzM3MjE3MzoxMjY2NTc3&notifications_query=is%3Aunread#issuecomment-2111622804

Feel free to reopen this issue if you find a different solution or if the nature of macOS signing allows a more fine grained control. If you prefer, I close this as "not planned" indicating that I don't think that there's anything for us to do.

**Comment by mikehearn** [comment]
The identifier in the "background items" list in Settings doesn't have to use the signing identity. Other items in there don't. You can set the AssociatedBundleIdentifiers key in the plist when registering it to the bundle ID of an application, and that should show up instead:

https://stackoverflow.com/questions/75918835/how-to-adjust-what-is-displayed-in-the-login-items-list-for-my-launch-daemon-age

If you have a bunch of background items already, you can probably see some that aren't using signing identities and take a look at how they do it. macOS falls back to signing identity only when it doesn't have anything better.

**Comment by mikehearn** [comment]
Note that Apple documents the above here:

https://developer.apple.com/documentation/servicemanagement/updating-helper-executables-from-earlier-versions-of-macos

**Comment by fgeorgatos** [comment]
I still have an itch with this case.

Here's my beef:
* someone is happy to come forward and donate their certificate to sign an open source project, so far so good.
* the same someone hasn't been a member of the project since ever, nor bothered with documenting a single commit.
* nor a message/communication by said person somewhere on what's the background story of how their key is entrusted.

@TWilb Am I missing something on how open source software communities are meant to go in terms of transparency?

**Comment by frankmorgner** [comment]
There will likely be changes in the near future, see #3512

**Comment by KlevsKalasec** [comment]
Have the same issue, this is outrageous. whoever he is
